motor/discovery/aws/resolver.go

```diff
@@ -107,7 +107,7 @@ func (r *Resolver) Resolve(ctx context.Context, root *asset.Asset, tc *providers
 		log.Debug().Int("instances", len(assetList)).Msg("completed ssm instance search")
 		for i := range assetList {
 			a := assetList[i]
-			if resolvedRoot != nil {
+			if resolvedRoot != nil && a != nil {
```

Comment:
This looks like a good improvement, but we should figure out why we have empty assets in the list in the first place. We should not have nil assets in the asset list.